services/speechsynth.c

```diff
@@ -0,0 +1,123 @@
+#include "jd_services.h"
+#include "interfaces/jd_pins.h"
+#include "jacdac/dist/c/speechsynthesis.h"
+
+#define SPEAK_CMD 0x01
+#define CANCEL_CMD 0x02
+#define SET_LANG 0x04
+#define SET_RATE 0x08
+#define SET_PITCH 0x10
+#define SET_VOLUME 0x10
+static char phrase[200];
+
+struct srv_state {
+    SRV_COMMON;
+    const speech_synth_api_t *api;
+    char lang[20];
+    uint32_t pitch;
+    uint32_t rate;
+    uint8_t enabled;
+    uint8_t volume;
+    uint8_t flags;
+};
+
+REG_DEFINITION(                                   //
+    speech_synth_regs,                                   //
+    REG_SRV_COMMON,                         //
+    REG_U32(JD_REG_PADDING),
+    REG_BYTES(JD_REG_PADDING,20),                 // don't know how to do strings properly yet
```

Comment:
Please show me how to do strings @mmoskal !

Reply:
REG_BYTES() would kind of work, except it doesn't do NUL-termination (either way). Would need to add this.